src/sentry/api/endpoints/organization_events_root_cause_analysis.py

```diff
@@ -1,10 +1,74 @@
+from datetime import datetime
+
 from rest_framework.response import Response
+from snuba_sdk import Column, Function
 
 from sentry import features
 from sentry.api.api_publish_status import ApiPublishStatus
 from sentry.api.base import region_silo_endpoint
 from sentry.api.bases.organization_events import OrganizationEventsEndpointBase
+from sentry.search.events.builder import QueryBuilder
+from sentry.search.events.types import QueryBuilderConfig
+from sentry.snuba.dataset import Dataset
 from sentry.snuba.metrics_performance import query as metrics_query
+from sentry.utils.snuba import raw_snql_query
+
+
+def query_spans(transaction, regression_breakpoint, params):
+    selected_columns = [
+        "count(span_id) as span_count",
+        "sumArray(spans_exclusive_time) as total_span_self_time",
+        "array_join(spans_op) as span_op",
+        "array_join(spans_group) as span_group",
+        # want a single event id to fetch from nodestore for the span description
+        "any(id) as sample_event_id",
+    ]
+
+    builder = QueryBuilder(
+        dataset=Dataset.Discover,
+        params=params,
+        selected_columns=selected_columns,
+        equations=[],
+        query=f"transaction:{transaction}",
+        orderby=["span_op", "span_group", "total_span_self_time"],
+        limit=10000,
+        config=QueryBuilderConfig(
+            auto_aggregations=True,
+            use_aggregate_conditions=True,
+            functions_acl=[
+                "array_join",
+                "sumArray",
+                "percentileArray",
+            ],
+        ),
+    )
+
+    builder.columns.append(
+        Function(
+            "if",
+            [
+                Function(
+                    "greaterOrEquals", [Column("timestamp"), regression_breakpoint.isoformat()]
+                ),
+                "1",
+                "0",
+            ],
+            "period",
+        )
+    )
+    builder.columns.append(Function("countDistinct", [Column("event_id")], "transaction_count"))
+    builder.groupby.append(Column("period"))
+
+    snql_query = builder.get_snql_query()
+    results = raw_snql_query(snql_query, "api.organization-events-root-cause-analysis")
+
+    data = results.get("data", [])
+
+    # sumArray is not allowed in equations so we have to calculate this manually
+    for row in data:
+        row["average_span_duration"] = row["total_span_self_time"] / row["span_count"]
+
+    return data
```

Comment:
Is the intention to return these results unformatted and let the frontend handle the rest? The data is quite difficult to understand without context.

Reply:
@Zylphrex This is meant to be fed into the span analysis helper and the results of that will be massaged into something more readable, e.g. https://www.notion.so/sentry/Analysis-703a5c0b357b4b93a28802ae90591fb5?pvs=4#13862c1ced244a51a11a247e5245e389

There are a few iterations to continue adding but it should surface the changes in duration or count